Client/Frontend/Browser/BrowserViewController/BVC+Rewards.swift

```diff
@@ -115,18 +115,41 @@ extension Tab {
     func reportPageLoad(to rewards: BraveRewards) {
         guard let webView = webView, let url = webView.url else { return }
         if url.isLocal || PrivateBrowsingManager.shared.isPrivateBrowsing { return }
+                
+        var htmlBlob: String?
+        var classifierText: String?
         
-        let getHtmlToStringJSCall = "document.documentElement.outerHTML.toString()"
-        // Copy to var, as `shouldClassifyLoadsForAds` can be reset before JS completes
-        let shouldClassify = shouldClassifyLoadsForAds
-        webView.evaluateJavaScript(getHtmlToStringJSCall, completionHandler: { html, _ in
+        let group = DispatchGroup()
+        group.enter()
+        webView.evaluateJavaScript("document.documentElement.outerHTML.toString()", completionHandler: { html, _ in
+            defer { group.leave() }
             guard let htmlString = html as? String else { return }
+            htmlBlob = htmlString
+        })
+        
+        if shouldClassifyLoadsForAds {
+            group.enter()
+            webView.evaluateJavaScript("document.body.innerText", completionHandler: { text, _ in
+                defer { group.leave() }
+                guard var innerText = text as? String else { return }
+                for c in ["\'", "\"", "\\", "\n", "\r", "\t", "\0"] {
+                    innerText = innerText.replacingOccurrences(of: c, with: " ")
```

Comment:
Thinking whether this can be a performance problem for large websites since we run this multiple times, perhaps a regular expression or other refactoring could be done here?
I'm fine with current solution for now, we can iterate on it later

Reply:
I'll see if I can update this with a regex pattern that will match against all of these. This is just the simplest solution, if performance is a concern we could also throw it on a bg thread since we're already using dispatch groups